libvips/foreign/jxlload.c

```diff
@@ -582,7 +571,8 @@ vips_foreign_load_jxl_generate(VipsRegion *out_region,
 
 	g_assert(r->height == 1);
 
-	if (vips_foreign_load_jxl_read_frame(jxl, jxl->frame, frame))
+	if (jxl->frame_no < frame &&
```

Comment:
Ah I see, you've moved the check outside the function. Would it be safer inside?

Reply:
Commit 4c8dde602d14171c57d466ea97c24f3348c7b166 moves it back inside the function.